operator/internal/status/lokistack.go

```diff
@@ -87,6 +93,45 @@ func generateCondition(cs *lokiv1.LokiStackComponentStatus) metav1.Condition {
 		len(cs.Ruler[corev1.PodPending])
 
 	if pending != 0 {
+		// TODO if condition is pending then look at the nodes if the labels match the topologyKey
+		if stack.Spec.Replication != nil && len(stack.Spec.Replication.Zones) > 0 {
+			podList := &corev1.PodList{}
+			if err := k.List(ctx, podList, &client.ListOptions{
+				Namespace: stack.Namespace,
+				LabelSelector: labels.SelectorFromSet(labels.Set{
+					"app.kubernetes.io/instance": stack.Name,
+				}),
+			}); err != nil {
+				log.Error(err, "Error getting pod resources")
+				return conditionPending
+			}
+
+			for _, pod := range podList.Items {
+				if pod.Status.Phase == corev1.PodPending {
+					nodeList := corev1.NodeList{}
+					if err := k.List(ctx, &nodeList, &client.ListOptions{
+						LabelSelector: labels.SelectorFromSet(labels.Set{
+							lokiv1.AnnotationAvailabilityZoneLabels: pod.Annotations[lokiv1.AnnotationAvailabilityZoneLabels],
+						}),
+					}); err != nil {
+						log.Error(err, "Error getting nodes")
+						return conditionPending
+					}
+
+					if len(nodeList.Items) == 0 {
+						degradedError := DegradedError{
+							Message: "Availability zone pod annotation does not match it's node's labels",
+							Reason:  lokiv1.ReasonAvailabilityZoneLabelsMismatch,
+							Requeue: false,
+						}
+						if err := SetDegradedCondition(ctx, k, req, degradedError.Message, degradedError.Reason); err != nil {
```

Comment:
No need to call to `SetDegradedCondition` as we're already at the point of "setting conditions" here. Simply return the degraded `Condition` (not `DegradedError`) instead of the pending condition.